python/ray/data/_internal/stage_impl.py

```diff
@@ -0,0 +1,194 @@
+from typing import Optional, TYPE_CHECKING
+
+import ray
+from ray.data._internal.fast_repartition import fast_repartition
+from ray.data._internal.plan import AllToAllStage
+from ray.data._internal.shuffle_and_partition import (
+    PushBasedShufflePartitionOp,
+    SimpleShufflePartitionOp,
+)
+from ray.data._internal.block_list import BlockList
+from ray.data._internal.remote_fn import cached_remote_fn
+from ray.data._internal.sort import sort_impl
+from ray.data.context import DatasetContext
+from ray.data.block import (
+    _validate_key_fn,
+    Block,
+    KeyFn,
+    BlockMetadata,
+    BlockAccessor,
+    BlockExecStats,
+)
+
+if TYPE_CHECKING:
+    from ray.data import Dataset
+
```

Comment:
Nice refactoring. Shall we also pull out the one-to-one stages?

Reply:
I was going to do that later (since OneToOne stages are basically all the same, having them as explicit different classes is not that useful for optimization purposes).